### PR TITLE
implement username change

### PR DESCRIPTION
**Pull request by codyzu**
issue #79

### CODE REVIEW

**Comment by codyzu** [comment]
@gr2m ready!

*note: I had to refactor the mocks in tests to allow me to sometimes return a user doc with a different username... Let me know if we can/should improve this.*

File: tests/integration/routes/account/patch-account-test.js [hunk 1]
```diff
@@ -155,6 +254,21 @@ getServer(function (error, server) {
...
+    group.test('username change', function (t) {
+      var clock = lolex.install(0, ['Date'])
+      var couchdb = mockUsernameChange()
+      server.inject(usernameChangeOptions, function (response) {
+        clock.uninstall()
+
+        t.is(couchdb.pendingMocks()[0], undefined, 'all mocks satisfied')
+
+        t.is(response.statusCode, 204, 'returns 204 status')
+        t.is(response.result, null, 'returns no body')
```
Comment (gr2m) [inline tests/integration/routes/account/patch-account-test.js]: can we check for the `x-set-session` response header here as implemented in https://github.com/hoodiehq/hoodie-account-server/issues/80?

**Comment by gr2m** [comment]
great work, just a minor comment: https://github.com/hoodiehq/hoodie-account-server/pull/155#discussion_r62904410. Other than that: LGTM!

ping @hoodiehq/maintainers

**Comment by codyzu** [comment]
@gr2m good idea... done!
I used this tonic to calculate the sessionId: https://tonicdev.com/57338e6682e0151100b86db5/57338e6682e0151100b86db6

**Comment by gr2m** [comment]
Many of our maintainers are conferencing / traveling right now, so I gave this another look and tested it locally with `hoodie-account` – it all worsk great. Thanks @codyzu, this will soon be released as `v3.1.1`

**Pull request merged**